zerver/views/streams.py

```diff
@@ -347,6 +348,10 @@ def update_stream_backend(
     if message_retention_days is not None:
         if not user_profile.is_realm_owner:
             raise OrganizationOwnerRequiredError()
+
+        if sub is None and stream.invite_only:
+            raise JsonableError(_("Invalid stream ID"))
```

Comment:
Feels this is a simple condition to understand and also avoids query used in `access_stream_by_id`.

Reply:
One other way can be to call `check_basic_stream_access`, but since we only want to have check for private stream, this is simple and easy to understand for someone who is reading the code.

Reply:
One advantage of using existing function will be when we add some future checks for the permission to access stream in general. But yeah, these checks might be change as per the new groups permissions model, so not sure worth thinking about much.

Reply:
I don't like having what feel like random checks for specific settings -- we want everything to feel like part of a coherent system, and checks like this one scary me as possible security bugs waiting to happen.

I think we can go with either this or `check_basic_stream_access`, but in either case we need a block comment explaining what we're doing. 

An alternative theory is we should just pause on this until we've added a "Who can administer this stream group" setting, since that's going to be much more impactful and may inform what organization we want here.